src/wp-includes/option.php

```diff
@@ -776,21 +776,13 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
-	/*
-	 * If the new and old values are the same, no need to update.
-	 *
-	 * Unserialized values will be adequate in most cases. If the unserialized
-	 * data differs, the (maybe) serialized data is checked to avoid
-	 * unnecessary database calls for otherwise identical object instances.
-	 *
-	 * See https://core.trac.wordpress.org/ticket/38903
-	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	/** This filter is documented in wp-includes/option.php */
+	$default_value = apply_filters( "default_option_{$option}", false, $option, false );
+
+	if ( $old_value !== $default_value && _is_equal_database_value( $old_value, $value ) ) {
```

Comment:
I think it's better not to remove the comment here. We should just slightly alter it.
```suggestion
	/*
	 * If the new and old values are the same, no need to update.
	 *
	 * An exception applies when no value is set in the database, i.e. the old value is the default.
	 * In that case, the new value should always be added as it may be intentional to store it rather than relying on the default.
	 *
	 * See https://core.trac.wordpress.org/ticket/38903 and https://core.trac.wordpress.org/ticket/22192.
	 */
	if ( $old_value !== $default_value && _is_equal_database_value( $old_value, $value ) ) {
```